pages/account/home.js

```diff
@@ -9,56 +9,93 @@ import { errorsPropType } from '../../services/propTypes'
+  const [confirmedCompanies, setConfirmedCompanies] = useState([])
+  const [company, setCompany] = useState([])
+
+  const setCompanyDataAndProfile = () => {
+    if (sessionStorage) {
+      const storageCompanyData = JSON.parse(sessionStorage?.getItem('companyData'))
+      if (storageCompanyData) {
+        setCompanyData(storageCompanyData)
+      }
+
+      const storageProfile = JSON.parse(sessionStorage?.getItem('profile'))
+      if (storageProfile) {
+        setProfile(storageProfile)
+      }
 
-  const confirmedCompanies = companyData.filter((company) => company.membershipStatus === 'confirmed')
-  const pendingCompanies = companyData.filter((company) => company.membershipStatus === 'pending')
+      const filteredConfirmedCompanies = companyData.filter(
+        (company) => company.membershipStatus === 'confirmed'
+      )
+      setConfirmedCompanies(filteredConfirmedCompanies)
 
-  const companyMatch = companyData.filter((company) => company.number === companyNo ? company : false)
-  const company = companyMatch[0]
+      const filterPendingCompanies = companyData.filter(
+        (company) => company.membershipStatus === 'pending'
+      )
+      setPendingCompanies(filterPendingCompanies)
 
-  React.useEffect(() => {
+      const companyMatch = companyData.filter((company) =>
+        company.number === companyNo ? company : false
+      )
+      const filteredCompany = companyMatch[0] || []
+      setCompany(filteredCompany)
+    }
+  }
+
+  useEffect(() => {
     headingCount.reset()
   })
 
+  useEffect(() => {
+    setCompanyDataAndProfile()
+    setLoading(false)
+  })
+
   return (
     <FeatureDynamicView
-      width="full"
-      titleLinkHref="/account/home"
+      width='full'
+      titleLinkHref='/account/home'
       hasBackLink={false}
       hasLanguageSwitcher={true}
       hasLogoutLink={true}
       hasAccountLinks={true}
       accountLinksItem={1}
-      messages={pendingCompanies.length}
+      messages={pendingCompanies ? pendingCompanies.length : ''}
     >
-
       {loading
-        ? <Loading/>
-        : <Dynamic
-        componentMap={componentMap}
-        headingCount={headingCount}
-        content={content}
-        errors={errors}
-        uiElements={[]}
-        uiStage={uiStage}
-        profile={profile}
-        companyData={{ count: confirmedCompanies.length, pendingCount: pendingCompanies.length }}
-        company={company ? mapCompanyData(company) : null}
-        links={{ ewfAuthenticatedEntry: CH_EWF_AUTHENTICATED_ENTRY_URL }}
-        {...router.query}
-      />}
+        ? (<Loading />)
+        : (
+        <Dynamic
+          componentMap={componentMap}
+          headingCount={headingCount}
+          content={content}
+          errors={errors}
+          uiElements={[]}
+          uiStage={uiStage}
+          profile={profile}
+          companyData={{
+            count: confirmedCompanies ? confirmedCompanies.length : 0,
```

Comment:
```suggestion
            count: confirmedCompanies ? confirmedCompanies.length : 0,
            pendingCount: pendingCompanies ? pendingCompanies.length : 0
```
These will always be 0 as they come from the company data if you trace it back.
We need to think on an approach going forward

Reply:
Ref - 0 companies despite having companies attached to account:
![image](https://user-images.githubusercontent.com/46429371/233116000-5a50318f-6252-4227-9668-69a13da8faf4.png)